TAF_EasyRest/TestFramework/PageComponents/BasePageComponent.cs

```diff
@@ -0,0 +1,29 @@
+﻿using TestFramework.Tools;
+
+namespace TestFramework.PageComponents
+{
+    public class BasePageComponent
+    {
+        protected IWebElement WaitUntilElementIsVisible(By locator, IWebElement parent, int timeToWait)
+        {
+            var driver = parent.GetWebDriverFromWebElement();
+
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(timeToWait));
+            var el = wait.Until(ExpectedConditions.ElementIsVisible(locator));
+            return el;
```

Comment:
You can use the `WaitUntilElementIsVisible(By locator, IWebDriver driver, int timeToWait)` method here to not repeat the code.

Reply:
done